src-electron/generator/matter/darwin/Framework/CHIP/templates/helper.js

```diff
@@ -240,12 +240,14 @@ async function asObjectiveCClass(type, cluster, options) {
   }
 
   if (isStruct) {
-    const structObj = await zclQuery.selectStructByName(
+    const structObj = await zclQuery.selectStructByNameAndClusterName(
       this.global.db,
       type,
+      cluster,
       pkgIds
     );
-    if (structObj.structClusterCount == 0) {
+
+    if (!structObj.clusterName) {
       // This is a global struct.
```

Comment:
Here's where I think we should handle the cases when "cluster" is not an actual cluster name.  Something like:
```suggestion
    let isGlobalStruct;
    if (!structObj) {
      // This can fail when our "cluster name" is a backwards compat name.  Just
      // fetch by struct name only in that case.
      structObj = await zclQuery.selectStructByName(
        this.global.db,
        type,
        pkgIds
      );

      isGlobalStruct = (structObj.structClusterCount == 0);
    } else {
      isGlobalStruct = !structObj.clusterName;
    }

    if (isGlobalStruct) {
      // This is a global struct.
```

though the fact that the two APIs have totally different ways of representing global structs is kind of annoying.

Reply:
Had to change it because of the extended use case of same named data types for global vs cluster specific ones. Is there anything you want back? count logic instead?

Reply:
added the suggestion